push_notifications/webpush.py

```diff
@@ -64,7 +64,7 @@ def webpush_send_message(device, message, results=None, **kwargs):
 				 "original_registration_id": device.registration_id})
 		return results
 	except WebPushException as e:
-		if "<Response [410]>" in e.message or "NotRegistered" in e.message or "InvalidRegistration" in e.message or "UnauthorizedRegistration" in e.message:
+		if "<Response [410]>" in e.message or "NotRegistered" in e.message or "InvalidRegistration" in e.message or "UnauthorizedRegistration" in e.message or "InvalidTokenFormat" in e.message:
```

Comment:
Squash this into the commit that introduces that line.

Also that line should look like this:

```
known_errors = ("NotRegistered", "InvalidRegistration", ...)
if e.message in known_errors:
   ...
```

The "<Response [410]>" doesn't pass the smell test either, sounds like you should be checking the http code.

Reply:
The last commit change these lines for:

    controlled_errors = (
        "<Response [410]>", "NotRegistered", "InvalidRegistration",
        "UnauthorizedRegistration", "InvalidTokenFormat")
    if any(controlled_error in e.message for controlled_error in controlled_errors):

Your lines does not work because e.message is not, for example, "NotRegistered". e.message is "Not Registered bla bla bla bla bla".

Yes, "<Response [410]>"  is ugly. But when a firefox device is outdate, it is the response.

We was recollect from July until now the error messages...